driver-core/src/main/com/mongodb/client/model/expressions/NumberExpression.java

```diff
@@ -21,4 +21,39 @@
  */
 public interface NumberExpression extends Expression {
 
+    // TODO this must not return <T extends NumberExpression> T
+    // since it allows: IntegerEx result = oneNum.multiply(oneInt)
+    NumberExpression multiply(NumberExpression n);
```

Comment:
I previously suggested that these should return `<T extends NumberExpression> T`, but that allows us to write `IntegerEx result = oneNum.multiply(oneInt)`, which is incorrect.